core/src/main/java/org/elasticsearch/transport/netty/NettyTransportChannel.java

```diff
@@ -134,6 +141,14 @@ public void sendResponse(Throwable error) throws IOException {
         transportServiceAdapter.onResponseSent(requestId, action, error);
     }
 
+    private void close() {
```

Comment:
>  we are "closing" and then still proceeding to perform operations on the channel.

That's a pure implementation detail. The caller cannot observe this. It just observes that after the response is sent once, it cannot perform any "write" operation (sending a response) again.

> Can I suggest release or unreserve?

From the perspective of the channel it is still "closed", but if we really have to do this, then I'd more lean towards `release`.